eng/scripts/Copy-ApiVersion.ps1

```diff
@@ -1,77 +1,151 @@
+                $validProviders = (Get-ChildItem "$PSScriptRoot/../../specification/$ServiceDirectory/$ServiceType/" -Directory).Name -join ', '
+                throw "Service Provider not found. Valid options are: $validProviders"
             }
+            $true
+        })]
+    [ArgumentCompleter({
+            param($commandName,
+                $parameterName,
+                $wordToComplete,
+                $commandAst,
+                $fakeBoundParameters)
+            Get-ChildItem "$PSScriptRoot/../../specification/$($fakeBoundParameters.ServiceDirectory)/$($fakeBoundParameters.ServiceType)/" -Directory | Select-Object -ExpandProperty Name
+        })]
+    [string] $Provider,
+
+    [Parameter(Mandatory = $true)]
+    [ArgumentCompleter({ "preview/", "stable/" })]
+    [ValidateScript({ 
+            function script:Get-Version([string] $version) {
+                if ($version -match '^(?<status>stable|preview)/(?<version>(\d{4}-\d{2}-\d{2}|\d+\.\d+)(-preview(\.\d+)?)?)$') {
+                    return $Matches['status'], $Matches['version']
+                }
         
-            throw 'Version must start with "stable/" or "preview/" and end with either a date-based version (recommended) like 2024-01-05 or a major.minor semver, followed by an optional "-preview" for previews APIs.'
-        }
-        # Throwing or returning truthy is sufficient.
-        Get-Version $_
-    })]
-    [string] $FromVersion,
-
-    [Parameter()]
-    [ValidateScript({Get-Version $_})]
-    [string] $ToVersion
+                throw 'Version must start with "stable/" or "preview/" and end with either a date-based version (recommended) like 2024-01-05 or a major.minor semver, followed by an optional "-preview" for previews APIs.'
+            }
+            Get-Version $_ })]
+    [string] $NewVersion,
+    
+    [Parameter(Mandatory = $true)]
+    [ValidateScript({ Get-Version $_ })]
+    [ArgumentCompleter({
+            param($commandName,
+                $parameterName,
+                $wordToComplete,
+                $commandAst,
+                $fakeBoundParameters)
+            [string[]] $stableVersions = Get-ChildItem "$PSScriptRoot/../../specification/$($fakeBoundParameters.ServiceDirectory)/$($fakeBoundParameters.ServiceType)/$($fakeBoundParameters.Provider)/stable/" -Directory | Select-Object -ExpandProperty Name |
+            ForEach-Object { "stable/$_" }
+            [string[]] $previewVersions = Get-ChildItem "$PSScriptRoot/../../specification/$($fakeBoundParameters.ServiceDirectory)/$($fakeBoundParameters.ServiceType)/$($fakeBoundParameters.Provider)/preview/" -Directory | Select-Object -ExpandProperty Name |
+            ForEach-Object { "preview/$_" }
+            return $stableVersions + $previewVersions
+        })]
+    [string] $BaseVersion,
+
+    [string] $ProviderInnerFolder = $null,
+    
+    [bool] $testMode = $false
```

Comment:
You should use the WhatIf/ShouldProcess workflow already in PS https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.4